code/src/java/pcgen/gui2/facade/Gui2InfoFactory.java

```diff
@@ -859,15 +859,15 @@ private HtmlInfoBuilder getEquipmentHtmlInfo(Equipment equip)
 
 		Integer a = EqToken.getMaxDexTokenInt(pc, equip);
 
-		if (a.intValue() != Constants.MAX_MAXDEX)
```

Comment:
new Integer(3) == new Integer(3) returns false.
So if a is an object (Integer) and you remove the .intValue() call then in the future if Constants.MAX_MAXDEX is changed an object, then the behavior of this method changes, as the two values are tested for instance equality, not value equality.  So there is a case with automated unboxing that actually is higher risk of producing problems, which is why I point is out.  The point here is that the .intValue() in this case is not actually useless.  It is, in fact, defensive programming, and in my opinion should be left alone.